Source/objects.cpp

```diff
@@ -451,16 +451,14 @@ void AddCandles()
 void AddBookLever(Rectangle affectedArea, _speech_id msg)
 {
 	int cnt = 0;
-	int xp;
-	int yp;
+	Point position;
 	bool exit = false;
 	while (!exit) {
 		exit = true;
-		xp = GenerateRnd(80) + 16;
-		yp = GenerateRnd(80) + 16;
+		position = Point(GenerateRnd(80), GenerateRnd(80)) + Displacement { 16, 16 };

```

Comment:
Pretty sure this is indeterminately sequenced due to calling the constructor function-style

>Every expression in a comma-separated list of expressions in a parenthesized initializer is evaluated as if for a function call (indeterminately-sequenced)

You can ensure the roll for x happens before the roll for y by using list initialisation:
>In [list-initialization](https://en.cppreference.com/w/cpp/language/list_initialization), every value computation and side effect of a given initializer clause is sequenced before every value computation and side effect associated with any initializer clause that follows it in the brace-enclosed comma-separated list of initalizers.
```suggestion
		position = Point { GenerateRnd(80), GenerateRnd(80) } + Displacement { 16, 16 };
```